src/js/components/AppListComponent.jsx

```diff
@@ -296,6 +300,7 @@ var AppListComponent = React.createClass({
       appsStatusesCount: getInitialFilterCounts(AppStatus),
       appsHealthCount: getInitialFilterCounts(HealthStatus)
     };
+    var filterText = props.filters[FilterTypes.TEXT];
```

Comment:
wait. the filterText value is in the condition of the `if` which is checked anyway every time. why should i miss one "cache" hit?